File: src/core/validators/EnumerationProperty/EnumerationPropertyMustMatchAnEnumeration.cs
using MetaEd.Grammar.Antlr;

namespace MetaEd.Core.Validator.EnumerationProperty
{
    public class EnumerationPropertyMustMatchAnEnumeration : ValidationRuleBase<MetaEdGrammar.EnumerationPropertyContext>
    {
        private readonly ISymbolTable _symbolTable;

        public EnumerationPropertyMustMatchAnEnumeration(ISymbolTable symbolTable)
        {
            _symbolTable = symbolTable;
        }

        public override bool IsValid(MetaEdGrammar.EnumerationPropertyContext context)
        {
            var identifierToMatch = context.propertyName().GetText();
            return _symbolTable.IdentifierExists(SymbolTableEntityType.EnumerationEntityType(), identifierToMatch);
        }

        public override string GetFailureMessage(MetaEdGrammar.EnumerationPropertyContext context)
        {
            return string.Format("Enumeration property '{0}' does not match any declared enumeration.", context.propertyName().GetText());
        }
    }
}

File: src/core/validators/EnumerationProperty/EnumerationPropertyMustMatchAnEnumeration.ts
import { ValidationRuleBase } from "../ValidationRuleBase";
import {ISymbolTable} from '../SymbolTable'
export class EnumerationPropertyMustMatchAnEnumeration extends ValidationRuleBase<MetaEdGrammar.EnumerationPropertyContext>
{
    private symbolTable: ISymbolTable;
    constructor(symbolTable: ISymbolTable) {
        super();
        this.symbolTable = symbolTable;
    }
    public isValid(context: MetaEdGrammar.EnumerationPropertyContext): boolean {
        let identifierToMatch = context.propertyName().GetText();
        return this.symbolTable.identifierExists(SymbolTableEntityType.EnumerationEntityType(), identifierToMatch);
    }
    public getFailureMessage(context: MetaEdGrammar.EnumerationPropertyContext): string {
        return `Enumeration property '${context.propertyName().GetText()}' does not match any declared enumeration.`;
    }
}
